packages/@orbit/jsonapi/src/lib/query-operators.ts

```diff
@@ -12,108 +10,94 @@ import {
     const { record } = expression;
-    const requestOptions = customRequestOptions(source, query);
-    const settings = buildFetchSettings(requestOptions);
+    const requestOptions = requestProcessor.customRequestOptions(query);
+    const settings = requestProcessor.buildFetchSettings(requestOptions);
 
-    const document = await source.fetch(source.resourceURL(record.type, record.id), settings);
+    const document = await requestProcessor.fetch(requestProcessor.urlBuilder.resourceURL(record.type, record.id), settings);
 
-    const deserialized = source.serializer.deserialize(document);
-    const operations = operationsFromDeserializedDocument(deserialized);
+    const deserialized = requestProcessor.serializer.deserialize(document);
+    if (deserialized.meta) {
+      requestProcessor.processMeta(deserialized.meta, query);
+    }
+    const operations = requestProcessor.operationsFromDeserializedDocument(deserialized);
 
     const transforms = [buildTransform(operations)];
     const primaryData = deserialized.data;
 
     return { transforms, primaryData };
   },
 
-  async findRecords(source: JSONAPISource, query: Query): Promise<QueryOperatorResponse> {
+  async findRecords(requestProcessor: JSONAPIRequestProcessor, query: Query): Promise<QueryOperatorResponse> {
     const expression = query.expression as FindRecords;
     const { type } = expression;
-
-    let requestOptions: RequestOptions = {};
+    let { urlBuilder } = requestProcessor;
+    let standardRequestOptions: RequestOptions = {};
 
     if (expression.filter) {
-      requestOptions.filter = buildFilterParam(source, expression.filter);
+      standardRequestOptions.filter = await urlBuilder.buildFilterParam(expression.filter);
     }
 
     if (expression.sort) {
-      requestOptions.sort = buildSortParam(source, expression.sort);
+      standardRequestOptions.sort = await urlBuilder.buildSortParam(expression.sort);
     }
 
     if (expression.page) {
-      requestOptions.page = buildPageParam(source, expression.page);
+      standardRequestOptions.page = await urlBuilder.buildPageParam(expression.page);
     }
 
-    let customOptions = customRequestOptions(source, query);
+    let customOptions = requestProcessor.customRequestOptions(query);
+    let requestOptions = standardRequestOptions;
     if (customOptions) {
-      merge(requestOptions, customOptions);
+      requestOptions = mergeRequestOptions(standardRequestOptions, customOptions);
     }
 
-    const settings = buildFetchSettings(requestOptions);
+    const settings = requestProcessor.buildFetchSettings(requestOptions);
 
-    const document = await source.fetch(source.resourceURL(type), settings);
+    const document = await requestProcessor.fetch(requestProcessor.urlBuilder.resourceURL(type), settings);
 
-    const deserialized = source.serializer.deserialize(document);
-    const operations = operationsFromDeserializedDocument(deserialized);
```

Comment:
As per my suggestion above, we could call the following prior to deserialization:

```
requestProcessor.preprocessResponseDocument(document, query);
```